src/components/ResizablePanels.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <div class="resizable-panels">
+    <div class="resizable-panels__panel" :style="leftPanelWidth">
+      <slot name="left-panel">Left panel</slot>
+    </div>
+
+    <div class="resizable-panels__resizer" @mousedown="startResize()">
+      <div class="resizable-panels__line"></div>
+    </div>
+
+    <div class="resizable-panels__panel" :style="rightPanelWidth">
+      <slot name="right-panel">Right panel</slot>
+    </div>
+  </div>
+</template>
+<script  lang="ts">
+import Vue from 'vue';
+import { Component } from 'vue-property-decorator';
+
+@Component({
+  name: 'resizable-panels',
+})
+export default class ResizablePanels extends Vue {
+  private ratio: number = 0.4;
```

Comment:
You could let typescript infer the data type.

```suggestion
  private ratio = 0.4;
```